### PR TITLE
Optionally disable embedding JWK in protected header

### PR DESCRIPTION
**Pull request by boivie**
In some applications, the signer's public key is already known to
the recipient, or the 'kid' property in the header can be used to
lookup the key. Simply speaking, embedding the signer's public key
in the protected header's 'jwk' property is sometimes not wanted.

This commit adds a new method to the Signing interface to disable
embedding of the public key of the signer. It will be embedded by
default to maintain backwards compatibility.

### CODE REVIEW

File: signing.go [hunk 1]
```diff
@@ -31,12 +31,14 @@ type NonceSource interface {
...
 	Sign(payload []byte) (*JsonWebSignature, error)
 	SetNonceSource(source NonceSource)
+	EmbedJwk(embed bool)
 }
 
 // MultiSigner represents a signer which supports multiple recipients.
 type MultiSigner interface {
 	Sign(payload []byte) (*JsonWebSignature, error)
 	SetNonceSource(source NonceSource)
+	EmbedJwk(embed bool)
```
Comment (csstaub) [inline signing.go]: Rename this to `SetEmbedJwk`

**Comment by csstaub** [comment]
This looks good, but makes me feel we should probably refactor the Signer and Encrypter interfaces altogether. I think it would be preferable if signers/encrypters were immutable and options like this could only be set when building the initial object. I'll put this on my TODO list for a v2 maybe.

**Comment by boivie** [comment]
I have updated my commit to use SetEmbedJwk instead.

Regarding immutable signers, I do agree. I would also like to see a way to add per-message unique fields (such as "cty") in the (protected) header, which would also warrant an interface update. I guess v2 is just on the drawing board?

**Comment by csstaub** [comment]
Well, it's "on the drawing board" in the sense that "I don't like the way it is now and I'd like to clean it sometime". Also, per-message unique fields would be great! Another candidate for a v2 addition.

**Comment by csstaub** [comment]
Ah, can you also sign our CLA? See here: https://docs.google.com/a/squareup.com/forms/d/13WR8m5uZ2nAkJH41k7GdVBXAAbzDk00vxtEYjd6Imzg/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1

**Comment by boivie** [comment]
Certainly - I have signed the CLA

Thanks for merging the PR!

**Comment by csstaub** [comment]
Thanks, and thanks for your contribution!

**Pull request merged**